### ISSUE TITLE
Remove *versioned.Event (should not be pointer) from swagger spec

### ISSUE THREAD

**Issue by nikhiljindal** [kubernetes/kubernetes#24240]
Ref https://github.com/kubernetes/kubernetes/pull/24226

https://github.com/kubernetes/kubernetes/pull/23806 added *versioned.Event to swagger spec.
It should not be a pointer.

cc @smarterclayton

**Comment by nikhiljindal** [comment]
We are in the last few weeks of 1.3.
Will be good to get this fixed.

**Comment by smarterclayton** [comment]
It's on my list after the protobuf stuff lands.  I believe it's a bug in gorestful.

**Comment by roberthbailey** [comment]
@smarterclayton what's the status of this issue?

**Comment by smarterclayton** [comment]
Has not hit the top of the priority list.  The bug is probably in
go-restful, not treating an alias correctly when generating the swagger.

On May 26, 2016, at 1:45 AM, Robert Bailey <notifications@github.com> wrote:

@smarterclayton <https://github.com/smarterclayton> what's the status of
this issue?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/issues/24240#issuecomment-221783092>

**Comment by lavalamp** [comment]
We're going to focus on swagger for 1.4-- if it gets done before then, great, but definitely it's not blocking 1.3.

**Comment by smarterclayton** [comment]
This shouldn't be P1 anymore.

**Comment by dims** [comment]
Do we need to yank 1.4 tag on this?

**Comment by smarterclayton** [comment]
Figured what this is (and it actually causes problems for OpenAPI as well as Swagger 1.0).

First bug is that we pass *versioned.Event instead of versioned.Event in api_installer - opening a PR to fix.  Second bug is that go-restful shouldn't act braindead on getting a pointer - will spawn something upstream.

We *probably* want to backport this to 1.4 if we expect end users to generate openapi from our code - several tools I tried to used failed to work.
